client/main.lua

```diff
@@ -24,193 +23,219 @@ local currentGear = {
+    lib.requestAnimDict("clothingshirt")
+
+	TaskPlayAnim(cache.ped, "clothingshirt", "try_shirt_positive_d", 8.0, 1.0, -1, 49, 0, 0, 0, 0)
 end
+
 local function takeCoral(coral)
-    if Config.CoralLocations[currentDivingLocation.area].coords.Coral[coral].PickedUp then return end
-    local ped = PlayerPedId()
+    if Config.CoralLocations[currentDivingLocation.area].coords.coral[coral].pickedUp then
+        return
+    end
+
     local times = math.random(2, 5)
-    if math.random() > Config.CopsChance then callCops() end
-    FreezeEntityPosition(ped, true)
+
+    if math.random() > Config.CopsChance then
+        callCops()
+    end
+
+    FreezeEntityPosition(cache.ped, true)
+
     QBCore.Functions.Progressbar("take_coral", Lang:t("info.collecting_coral"), times * 1000, false, true, {
         disableMovement = true,
         disableCarMovement = true,
         disableMouse = false,
-        disableCombat = true,
+        disableCombat = true
     }, {
         animDict = "weapons@first_person@aim_rng@generic@projectile@thermal_charge@",
         anim = "plant_floor",
-        flags = 16,
+        flags = 16
     }, {}, {}, function() -- Done
-        Config.CoralLocations[currentDivingLocation.area].coords.Coral[coral].PickedUp = true
+        Config.CoralLocations[currentDivingLocation.area].coords.coral[coral].pickedUp = true
+
         TriggerServerEvent('qb-diving:server:TakeCoral', currentDivingLocation.area, coral, true)
-        ClearPedTasks(ped)
-        FreezeEntityPosition(ped, false)
+
+        ClearPedTasks(cache.ped)
+        FreezeEntityPosition(cache.ped, false)
     end, function() -- Cancel
-        ClearPedTasks(ped)
-        FreezeEntityPosition(ped, false)
+        ClearPedTasks(cache.ped)
+        FreezeEntityPosition(cache.ped, false)
     end)
 end
+
 local function setDivingLocation(divingLocation)
     if currentDivingLocation.area ~= 0 then
-        for k in pairs(Config.CoralLocations[currentDivingLocation.area].coords.Coral) do
+        for k in pairs(Config.CoralLocations[currentDivingLocation.area].coords.coral) do
             if Config.UseTarget then
-                exports['qb-target']:RemoveZone(k)
+                if next(zones) then
+                    exports.ox_target:removeZone(zone[k])
+                end
             else
-                if next(zones) then zones[k]:destroy() end
+                if next(zones) then
+                    zones[k]:remove()
+                end
             end
         end
     end
+
     currentDivingLocation.area = divingLocation
-    for _, blip in pairs(currentDivingLocation.blip) do if blip then RemoveBlip(blip) end end
-    local radiusBlip = AddBlipForRadius(Config.CoralLocations[currentDivingLocation.area].coords.Area, 100.0)
+
+    for _, blip in pairs(currentDivingLocation.blip) do
+        if blip then
+            RemoveBlip(blip)
+        end
+    end
```

Comment:
Expanding this one out actually isn't a bad idea, however, it can be streamlined a bit further.
```suggestion
    for _, blip in pairs(currentDivingLocation.blip) do
        if blip then RemoveBlip(blip) end
    end
```